src/init.lua

```diff
@@ -480,71 +484,63 @@ end
 	:::info
 ]=]
 
-function Store:LoadKeep(key: string, unReleasedHandler: UnReleasedHandler): Promise
+function Store:LoadKeep(key: string, unreleasedHandler: unreleasedHandler): Promise
 	local store = self._store
 
+	local validLoadMethods = {
+		ForceLoad = "ForceLoad",
+		-- maybe add "Ignore"?
+		Cancel = "Cancel",
+	}
+
 	if self._mock then
 		print("Using mock store!")
 	end
 
-	if unReleasedHandler == nil then
-		unReleasedHandler = function(_)
-			return "Ignore"
+	if unreleasedHandler == nil then
+		unreleasedHandler = function(_)
+			return validLoadMethods.ForceLoad
 		end
 	end
 
-	if type(unReleasedHandler) ~= "function" then
-		error("UnReleasedHandler must be a function")
+	if type(unreleasedHandler) ~= "function" then
+		error("unreleasedHandler must be a function")
 	end
 
-	local identifier = string.format(
-		"%s/%s%s",
-		self._store_info.Name,
-		if self._store_info.Scope ~= nil then self._store_info.Scope .. "/" else "",
-		key
-	)
-
-	if Keeps[identifier] then -- TODO: check if got rejected before returning cache
-		return Promise.resolve(Keeps[identifier])
-	elseif
-		self._cachedKeepPromises[identifier]
-		and self._cachedKeepPromises[identifier].Status ~= Promise.Status.Rejected
-		and self._cachedKeepPromises[identifier].Status ~= Promise.Status.Cancelled
-	then
-		return self._cachedKeepPromises[identifier]
+	local id = `{self._store_info.Name}/{self._store_info.Scope or ""}{self._store_info.Scope and "/" or ""}{key}`
+
+	if Keeps[id] then -- TODO: check if got rejected before returning cache
+		return Promise.resolve(Keeps[id])
+	elseif self._cachedKeepPromises[id] and self._cachedKeepPromises[id].Status ~= Promise.Status.Rejected and self._cachedKeepPromises[id].Status ~= Promise.Status.Cancelled then
+		return self._cachedKeepPromises[id]
 	end
 
 	local promise = Promise.new(function(resolve, reject)
 		local keep: Keep.KeepStruct = store:GetAsync(key) or {} -- support versions
 
-		local success = canLoad(keep)
+		local forceLoad = nil
 
-		local forceload = nil
+		if not canLoad(keep) and keep.MetaData.ActiveSession then
+			local loadMethod = unreleasedHandler(keep.MetaData.ActiveSession)
 
-		if not success and keep.MetaData.ActiveSession then
-			local loadMethod = unReleasedHandler(keep.MetaData.ActiveSession)
+			if not validLoadMethods[loadMethod] then
+				warn(`unreleasedHandler returned an invalid value, defaulting to {validLoadMethods.ForceLoad}`) -- TODO: Custom Error Class to fire to IssueSignal
 
-			if loadMethod ~= "Ignore" and loadMethod ~= "Cancel" then
-				warn("UnReleasedHandler returned an invalid value, defaulting to Ignore") -- TODO: Custom Error Class to fire to IssueSignal
-
-				loadMethod = "Ignore"
+				loadMethod = validLoadMethods.ForceLoad
 			end
 
-			if loadMethod == "Cancel" then
+			if loadMethod == validLoadMethods.Cancel then
 				reject(nil) -- should this return an error object?
 				return
```

Comment:
When ``Cancel`` is passed as the handler, we should probably check if the session that owns it is past deadlock time. I would probably assume this is intended behavior if I read the docs